gen/docdb

```diff
@@ -49,56 +49,46 @@ foreach my $resourceData ($data->getChildElements) {
 my $out_file_name = "$DIRECTORY/$::SERVICE_NAME";
 
 open FILE,">$out_file_name" or die "Cannot open $out_file_name: $! \n";
-print FILE JSON::XS->new->utf8->pretty->encode($outputData);
+print FILE JSON::XS->new->utf8->pretty->canonical->encode($outputData);
 close (FILE);
 perunServicesInit::finalize;
 
 ##############################################################################
 #   Only subs definitions down there
 ##############################################################################
 
-sub processResourceData {
-	my $resourceData = shift;
-	my $membersElement = ($resourceData->getChildElements)[1];
-	processResourceMembersData $membersElement;
-}
-
 sub processGroupData {
-	my $groupData = shift;
-	my $membersElement = ($groupData->getChildElements)[1];
-	my %groupAttributes = attributesToHash $groupData->getAttributes;
+	my ($resourceId, $groupId) = @_;
 
 	my $group = {
-								cn          => $groupAttributes{$A_GROUP_NAME},
-								description => $groupAttributes{$A_GROUP_DESCRIPTION},
-								memberuids  => processGroupsMembersData $membersElement, $groupAttributes{$A_GROUP_NAME},
+								cn          => $data->getGroupAttributeValue( group => $groupId, attrName => $A_GROUP_NAME ),
+								description => $data->getGroupAttributeValue( group => $groupId, attrName => $A_GROUP_DESCRIPTION ),
+								memberuids  => processGroupsMembersData($resourceId, $groupId),
 							};
 
 	push @{$outputData->{'groups'}}, $group;
 
 }
 
 sub processResourceMembersData {
-	my $membersElement = shift;
-	for my $memberData ($membersElement->getChildElements) {
-		my %memberAttributes = attributesToHash $memberData->getAttributes;
+	my $resourceId = shift;
+	for my $memberId ($data->getMemberIdsForResource(resource => $resourceId)){
 
 		my $member = {
-										mail   => $memberAttributes{$A_USER_MAIL},
-										uid    => $memberAttributes{$A_USER_LOGIN},
-										cn     => unidecode $memberAttributes{$A_USER_NAME},
+										mail   => $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_MAIL ),
+										uid    => $data->getUserFacilityAttributeValue( member => $memberId, attrName => $A_USER_LOGIN ),
+										cn     => unidecode $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_NAME ),
 								 };
 
 		push @{$outputData->{'members'}}, $member;
 	}
 }
 
 sub processGroupsMembersData {
-	my ($membersElement, $groupName) = @_;
+	my ($resourceId, $groupId) = @_;
 	my @groupMembers = ();
-	for my $memberData ($membersElement->getChildElements) {
-		my %memberAttributes = attributesToHash $memberData->getAttributes;
-		push @groupMembers, $memberAttributes{$A_USER_LOGIN};
+	foreach my $memberId($data->getMemberIdsForResourceAndGroup( resource => $resourceId, group => $groupId )) {
+		push @groupMembers, $data->getUserFacilityAttributeValue( member => $memberId, attrName => $A_USER_LOGIN );
 	}
```

Comment:
You can sort `@groupMembers` like this:
```perl
@groupMembers = sort @groupMembers;
```
before returning them. It will create consistent order in output. Since it contains only users logins, their order shouldn't affect the service, but it will enable us to have consistent and comparable data.

Reply:
added